Refactor flow cell SQL filter function

henrikstranneheim opened the pull request:
## Description

Refactor flow cell SQL filter function

### Added

- Refactor flow cell SQL filter function
- Test

### Changed

- Standardize function naming to use ´filter´ instead of get
- Sort imports in selected modules


### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b add_has_backup_filter -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

diitaz93 commented on cg/store/filters/status_flow_cell_filters.py:
```diff
@@ -1,17 +1,22 @@
 
 
-def get_flow_cell_by_name(flow_cells: Query, flow_cell_name: str, **kwargs) -> Query:
+def filter_flow_cells_by_has_backup(flow_cells: Query, **kwargs) -> Query:
```
Not sure if it would be useful to also add a function to filter flow cells that do NOT have a backup.

diitaz93 reviewed (approved):
Great! 👍

seallard reviewed (approved):
👍  I can't see that `filter_flow_cells_by_has_backup` is used anywhere yet, but I guess this is part of a bigger project

ChrOertlin commented on cg/store/api/find_business_data.py:
```diff
@@ -396,29 +412,29 @@ def get_metrics_entry_by_flow_cell_name_sample_internal_id_and_lane(
         ).first()
 
-    def get_flow_cell_by_name(self, flow_cell_name: str) -> Flowcell:
+    def filter_flow_cell_by_name(self, flow_cell_name: str) -> Flowcell:
```
I though we use the patterns: when returning a query we use `filter` and when returning objects we use `get`

so `get_flow_cell_by_name` would be the name to use here.

henrikstranneheim replied:
You are right! A S&R got out of hand - reverted 👍

ChrOertlin commented on cg/store/filters/status_flow_cell_filters.py:
```diff
@@ -51,7 +56,8 @@ def apply_flow_cell_filter(
-    GET_BY_NAME: Callable = get_flow_cell_by_name
-    GET_BY_NAME_SEARCH: Callable = filter_flow_cell_by_name_search
-    GET_WITH_STATUSES: Callable = filter_flow_cells_with_statuses
+    FILTER_BY_CASE: Callable = filter_flow_cells_by_case
+    FILTER_BY_HAS_BACKUP: Callable = filter_flow_cells_by_has_backup
+    FILTER_BY_NAME: Callable = filter_flow_cell_by_name
+    FILTER_BY_NAME_SEARCH: Callable = filter_flow_cell_by_name_search
+    FILTER_WITH_STATUSES: Callable = filter_flow_cells_with_statuses
```
nice!

henrikstranneheim replied:
👍

henrikstranneheim commented:
Realized we did not (yet) need general has_backup filter function, so I reverted that and changed scope to a pure refactoring instead. @diitaz93 @ChrOertlin @seallard ready for a new review..

diitaz93 reviewed (approved):
👍

henrikstranneheim commented:
### use flow cell query ✅ 

```
θ60° 9d [hiseq.clinical@hasta:~] [S_base] 19s $ cg backup fetch-flow-cell -d -f HJFW5DSXX
Running command /bin/dsmc q archive */*HJFW5DSXX*.gpg
Found archived files for PDC query: */*HJFW5DSXX*.gpg
Encryption key found: /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.key.gpg
Flow cell found: /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.tar.gz.gpg
```

henrikstranneheim commented:
Deployed to production:

```
Log deploy... done.
cg, version 51.2.4
```
